nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -40,25 +41,55 @@ int Pooling2DLayer::initialize(Manager &manager) {
     ml_logw("Warning: the length of previous layer dimension is one");
   }
 
+  if (pooling_type == PoolingType::global_max ||
+      pooling_type == PoolingType::global_average) {
+    if (pool_size[0] != in_dim.height() || pool_size[1] != in_dim.width()) {
+      ml_logw(
+        "[Pooling2D] global_max, global_average does not except pool size");
+      pool_size[0] = in_dim.height();
+      pool_size[1] = in_dim.width();
+    }
+
+    if (padding[0] != 0 || padding[1] != 0) {
+      ml_logw("[Pooling2D] global_max, global_average does not except padding");
+      padding[0] = 0;
+      padding[1] = 0;
+    }
+
+    if (stride[0] != 1 || stride[1] != 1) {
```

Comment:
if a user sets stride more than 1, it is better to give an error on initializing than eating that error and forcing the model to be different than intended by the user.

Reply:
Same applies for above checks as well. But you must distinguish between initialized values and user set values.

Reply:
It's just a matter of policy but returning an error as you mentioned.

> But you must distinguish between initialized values and user set values.

we cannot distinguish if this is defaulted value or user set value. eg) pool_size must be same as in_dim.height(), in_dim.width(). but it defaults to `0, 0`

If we were to separate classes between pooling type, this could become an hard error.

Reply:
As user is allowed to set these, then lets compare it with its default values, and give error if not default.

Reply:
Done